### PR TITLE
Fix category mapper

### PR DESCRIPTION
**Pull request by etiotto**
This PR completes support for the `CategoryMapper` operator when the input is a tensor of strings. It adds a end-to-end test in `TestCategoryMapper.cpp`  for testing a tensor of strings input.

### CODE REVIEW

**Comment by etiotto** [comment]
@jenkins-droid test this please

File: src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp [hunk 1]
```diff
@@ -707,22 +707,21 @@ class KrnlStrlenOpLowering : public ConversionPattern {
     Location loc = op->getLoc();
 
     // Get a symbol reference to the strlen function, inserting it if necessary.
-    ModuleOp parentModule = op->getParentOfType<ModuleOp>();
-    auto strlenRef = getOrInsertStrlen(rewriter, parentModule);
+    ModuleOp module = op->getParentOfType<ModuleOp>();
+    auto strlenRef = getOrInsertStrlen(rewriter, module);
```
Comment (AlexandreEichenberger) [inline src/Conversion/KrnlToLLVM/KrnlToLLVM.cpp:711]: nit: are we trying not to use `auto` when not too burdensome?

File: src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp [hunk 2]
```diff
@@ -231,6 +231,29 @@ struct ONNXCategoryMapperOpLowering : public ConversionPattern {
...
+  Value loadElement(Value memref, ValueRange loopInd, Type elementType,
+      int64_t rank, const KrnlBuilder &createKrnl) const {
+    Value inputElem;
+    TypeSwitch<Type>(elementType)
+        .Case<IntegerType>(
+            [&](IntegerType) { inputElem = createKrnl.load(memref, loopInd); })
+        .Case<StringType>([&](StringType stringType) {
+          OpBuilder &builder = createKrnl.getBuilder();
+          Location loc = createKrnl.getLoc();
+          auto zero = builder.create<arith::ConstantOp>(
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp:243]: We have a MathBuilder `  Value constant(Type type, double val) const;` that is a bit more compact

File: src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp [hunk 3]
```diff
@@ -231,6 +231,29 @@ struct ONNXCategoryMapperOpLowering : public ConversionPattern {
...
+              loc, builder.getIntegerAttr(builder.getIntegerType(64), 0));
+          auto memRefType = MemRefType::get(
+              {rank}, StringType::get(elementType.getContext()));
+          auto stringMemRef =
```
Comment (AlexandreEichenberger) [inline src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp:247]: Could add a quick KrnlBuilder for that op too.

**Comment by tungld** [comment]
@etiotto could you address Alex's comments please?

**Comment by etiotto** [comment]
> @etiotto could you address Alex's comments please?
Working on it.

File: src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp (hunk 2 above)
Comment (etiotto) [reply src/Conversion/ONNXToKrnl/ML/CategoryMapper.cpp]: done

**Review by tungld — APPROVED** [review]
LGTM!

BTW, I remember we agreed that we should avoid doing force-pushed since it is difficult to review requested changes.

**Comment by etiotto** [comment]
> BTW, I remember we agreed that we should avoid doing force-pushed since it is difficult to review requested changes.
Yes, sorry. My build got into a messy state after merging main.

**Pull request merged**